### PR TITLE
Optimized the options for number format

### PR DESCRIPTION
**Pull request by Indiana8000**
Aggregated the old config parameter:
- humanFriendly
- humanFriendlyDecimal
- formatNumber
- decimal

Into the new one:
- valueFormat (Available options: simpleCommas, humanFriendly, humanFriendlyCompact)
- valueDecimals (Number of fixed digits after floating point, negative means up to x digits)
- minMaxFormat (Available options: simpleCommas, humanFriendly, humanFriendlyCompact)
- minMaxDecimals (Number of fixed digits after floating point, negative means up to x digits)
- language (default browser language)

In addition I changed the order for config extend to first load the default and than the parameter on the constructor overwrite the defaults. Maybe this is worth an extra option?

I also splited the list of config parameters in the readme.md. With my next changes I will categorize this more.

### CODE REVIEW

File: docs/justgage.js [hunk 1]
```diff
@@ -289,8 +268,71 @@
...
     };
 
+    // Backward compatibility, map old config options
+    if(obj.config.valueFormat == null && obj.config.minMaxFormat == null) {
+      if(kvLookup("humanFriendly", config, dataset, false)) {
+        obj.config.valueFormat = "humanFriendly";
+        obj.config.minMaxFormat = "humanFriendly";
+        obj.config.valueDecimals = kvLookup("humanFriendlyDecimal", config, dataset, obj.config.valueDecimals);
+        obj.config.minMaxDecimals = kvLookup("humanFriendlyDecimal", config, dataset, obj.config.minMaxDecimals);
+        console.log("* justgage: Used depricated option humanFriendly. Update to the new valueFormat/minMaxFormat")
```
Comment (robertsLando) [inline docs/justgage.js:326]:
```suggestion
        console.log("* justgage: Used deprecated option `humanFriendly`. Update to the new `valueFormat/minMaxFormat`")
```

File: docs/justgage.js [hunk 2]
```diff
@@ -289,8 +268,71 @@
...
+      } else if(kvLookup("formatNumber", config, dataset, false)) {
+        obj.config.valueFormat = "simpleCommas";
+        obj.config.minMaxFormat = "simpleCommas";
+        obj.config.valueDecimals = kvLookup("decimals", config, dataset, obj.config.valueDecimals);
+        obj.config.minMaxDecimals = kvLookup("decimals", config, dataset, obj.config.minMaxDecimals);
+        console.log("* justgage: Used depricated option formatNumber. Update to the new valueFormat/minMaxFormat")
```
Comment (robertsLando) [inline docs/justgage.js:332]:
```suggestion
        console.log("* justgage: Used deprecated option `formatNumber`. Update to the new `valueFormat/minMaxFormat`")
```

**Review by robertsLando — CHANGES_REQUESTED** [review]
It looks to me this is much less user friendly to configure compared to the old one

**Comment by Indiana8000** [comment]
Added your changes and finally got eslint running on my PC.
For me the original options didn't allow to use different formats on the value and min/max as I like:
Screenshot 2022-12-27 at 15-26-48 Smart Monitoring

File: .github/workflows/build.yml [hunk 3]
```diff
@@ -5,6 +5,8 @@ on:
     branches: [ master ]
   pull_request:
     branches: [ master ]
+  workflow_dispatch:
+    branches: [ master ]
```
Comment (robertsLando) [inline .github/workflows/build.yml:9]: Not needed

**Review by robertsLando — CHANGES_REQUESTED** [review]
I'm still not convinced about this... Cannot you simply add some options to the already existing ones instead of do so much 'breaking changes' ? I know you added back compatibility but it looks way more difficult now then before to configure

File: .github/workflows/build.yml (hunk 3 above)
Comment (Indiana8000) [reply .github/workflows/build.yml]: ok, was just a test as eslint first didn't worked local.

Comment (Indiana8000) [reply .github/workflows/build.yml]: I'm fine if you reject this PR. I'm just learning the howl git thing.
From my view there are many more config options needed to be changed. I also thought about sub-config objects. But I think this is the wrong place to discus.

Comment (robertsLando) [reply .github/workflows/build.yml]: I'm fine about adding new features to the module but this module is widely used and introducing such changes is not something I like to do, I prefer an easier approach to keep it simple

**Pull request closed without merging**